test_runner/regress/test_readonly_node.py

```diff
@@ -187,25 +205,30 @@ def generate_updates_on_main(
 
             generate_updates_on_main(env, ep_main, i, end=100)
 
-            # Trigger GC
-            for shard, ps in tenant_get_shards(env, env.initial_tenant):
-                client = ps.http_client()
-                gc_result = client.timeline_gc(shard, env.initial_timeline, 0)
-                log.info(f"{gc_result=}")
+            trigger_gc_and_select(env, ep_static)
 
-                assert (
-                    gc_result["layers_removed"] == 0
-                ), "No layers should be removed, old layers are guarded by leases."
+            # Trigger Pageserver restarts
+            for ps in env.pageservers:
+                ps.stop()
+                time.sleep(3)
```

Comment:
Where a sleep is used, please include a comment explaining why it's needed.

I guess in this case the sleep period relates to LSN_LEASE_LENGTH, should it be expressed as a fraction of that?

Reply:
I was trying to get at least one lsn lease request failure due to connection (testing compute should retry). Added explaination for all `time.sleep` usage in dcc017a5a5632df8a29ee9575b957a5e34c8101b.